pkg/providers/subnet/subnet.go

```diff
@@ -190,8 +192,8 @@ func (p *DefaultProvider) UpdateInflightIPs(createFleetInput *ec2.CreateFleetInp
 	// Process the CreateFleetOutput to pull out all the fulfilled subnetIDs
 	var fleetOutputSubnets []string
 	if createFleetOutput != nil {
-		fleetOutputSubnets = lo.Compact(lo.Uniq(lo.Map(createFleetOutput.Instances, func(fleetInstance *ec2.CreateFleetInstance, _ int) string {
-			if fleetInstance == nil || fleetInstance.LaunchTemplateAndOverrides == nil || fleetInstance.LaunchTemplateAndOverrides.Overrides == nil {
+		fleetOutputSubnets = lo.Compact(lo.Uniq(lo.Map(createFleetOutput.Instances, func(fleetInstance ec2types.CreateFleetInstance, _ int) string {
+			if fleetInstance.InstanceIds == nil || fleetInstance.LaunchTemplateAndOverrides == nil || fleetInstance.LaunchTemplateAndOverrides.Overrides == nil {
```

Comment:
If it's not possible for fleetInstance to be nil anymore, we probably don't need this `fleetInstance.InstanceIds` nil check